docs/protocol-tutorial/0010-states-channels.md

```diff
@@ -0,0 +1,305 @@
+          fixedPart.chainId,
+          fixedPart.participants,
+          fixedPart.channelNonce,
+          fixedPart.appDefinition,
+          fixedPart.challengeDuration
+      )
+  );
+
+```
+
+## State commitments
+
+To commit to a state, a hash is formed as follows:
+
+```solidity
+ bytes32 stateHash = keccak256(abi.encode(
+        channelId,
+        variablePart.appData,
+        variablePart.outcome,
+        variablePart.turnNum,
+        variablePart.isFinal
+    ));
+```
+
+and this hash is signed using an _ephemeral_ Ethereum private key. _Ephemeral_ in this context means a dedicated private key, generated solely for the purpose of executing the state channel. The signature has the following type:
+
+=== "Solidity"
+
+    ```solidity
+        struct Signature {
+            uint8 v;
+            bytes32 r;
+            bytes32 s;
+        }
+    ```
+
+=== "TypeScript"
+
+    ```typescript
+        import { Signature } from "ethers";
+    ```
+
+=== "Go"
+
+    ```Go
+
+        type Signature struct {
+            R []byte
+            S []byte
+            V byte
+        }
+
+    ```
+
+You can sign a state using these utilities:
+
+=== "TypeScript"
+
+    ```typescript
+        import {signData} from '@statechannels/nitro-protocol;
+
+        const signature = signData(stateHash, privateKey);
+    ```
+
+=== "Go"
+
+    ```Go
+    import 	nc "github.com/statechannels/go-nitro/crypto"
+
+    signature := nc.SignEthereumMessage(stateHash.Bytes(), secretKey)
+    ```
+
+### `SignedVariableParts`
+
+Signatures on a state hash by different participants are often bundled up with the variable part of the state when being submitted to chain.
+
+=== "Solidity"
+
+    ```solidity
+    struct SignedVariablePart {
+        VariablePart variablePart;
+        Signature[] sigs;
+    }
+    ```
+
+=== "TypeScript"
+
+    ```typescript
+    export interface SignedVariablePart {
+        variablePart: VariablePart;
+        sigs: Signature[];
+    }
+
+    ```
+
+=== "Go"
+
+    ```Go
+        // INitroTypesSignedVariablePart is an auto generated low-level Go binding around an user-defined struct.
+        type INitroTypesSignedVariablePart struct {
+            VariablePart INitroTypesVariablePart
+            Sigs         []INitroTypesSignature
+        }
+    ```
+
+### Support proofs
+
+Support proofs consist of just enough information for the chain to verify that a state has been supported. They usually consist of [`FixedPart`](#states), plus a singular [`SignedVariablePart`](#signedvariableparts) named `candidate`, plus an array of [`SignedVariableParts`](#signedvariableparts) named `proof`.
+
+### `RecoveredVariableParts`
+
+The adjudicator smart contract will recover the signer from each signature on a `SignedVariablePart`, and convert the resulting list into a `signedBy` bitmask indicating which participant has signed that particular state. The bitfield is bundled with the `VariablePart` into a `RecoveredVariablePaet`:
```

Comment:
This is the first usage of the word adjudicator. Maybe it should be on top of page where the L1 lockup is mentioned (eg: locked up and managed by a contract we call the Adjudicator).

Reply:
[502b7fe](https://github.com/statechannels/go-nitro/pull/872/commits/502b7fe7bb01a71e0e1c8b07a5a3ef40a6f2eaa3)